common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java

```diff
@@ -0,0 +1,221 @@
+        } catch (BadPaddingException e) {
+            errCode = BAD_PADDING;
+            exception = e;
+        } catch (InvalidKeyException e) {
+            errCode = INVALID_KEY;
+            exception = e;
+        } catch (InvalidAlgorithmParameterException e) {
+            errCode = INVALID_ALG_PARAMETER;
+            exception = e;
+        }
+        throw new ClientException(errCode, exception.getMessage());
+    }
+
+    @Override
+    public byte[] decrypt(@NonNull final byte[] ciphertext) throws ClientException {
+        final String errCode;
+        final Exception exception;
+        try {
+            final SecretKeySpec key = new SecretKeySpec(this.key, suite.cipherName());
+            final Cipher c = Cipher.getInstance(key.getAlgorithm());
+            final IvParameterSpec ivSpec = new IvParameterSpec(ciphertext, 0, 12);
+            c.init(Cipher.DECRYPT_MODE, key, ivSpec);
+            byte[] out = Arrays.copyOfRange(ciphertext, 12, ciphertext.length);
+            return c.doFinal(out);
+        } catch (final NoSuchAlgorithmException e) {
+            errCode = NO_SUCH_ALGORITHM;
+            exception = e;
+        } catch (final NoSuchPaddingException e) {
+            errCode = NO_SUCH_PADDING;
+            exception = e;
+        } catch (IllegalBlockSizeException e) {
+            errCode = INVALID_BLOCK_SIZE;
+            exception = e;
+        } catch (BadPaddingException e) {
+            errCode = BAD_PADDING;
+            exception = e;
+        } catch (InvalidKeyException e) {
+            errCode = INVALID_KEY;
+            exception = e;
+        } catch (InvalidAlgorithmParameterException e) {
+            errCode = INVALID_ALG_PARAMETER;
+            exception = e;
+        }
+        throw new ClientException(errCode, exception.getMessage(), exception);
+    }
+
+    @Override
+    public byte[] sign(@NonNull final byte[] text, @NonNull final IDevicePopManager.SigningAlgorithm alg) throws ClientException {
+        final String errCode;
+        final Exception exception;
+        try {
+            final SecretKeySpec key = new SecretKeySpec(this.key, suite.cipherName());
+            Mac mac = Mac.getInstance(suite.macName());
+            mac.init(key);
+            return mac.doFinal(text);
+        } catch (final NoSuchAlgorithmException e) {
+            errCode = NO_SUCH_ALGORITHM;
+            exception = e;
+        } catch (InvalidKeyException e) {
+            errCode = INVALID_KEY;
+            exception = e;
+        }
+        throw new ClientException(errCode, exception.getMessage());
+    }
+
+    @Override
+    public boolean verify(@NonNull final byte[] text, @NonNull final IDevicePopManager.SigningAlgorithm alg,
+                          @NonNull final byte[] signature) throws ClientException {
+        return Arrays.equals(signature, sign(text, alg));
+    }
+
+    @Override
+    public byte[] getThumprint() throws ClientException {
+        return new byte[0];
```

Comment:
Would throwing be unacceptable here? Do we want to return empty `byte[]`?

Reply:
Throwing here would be awkward, I could return an empty byte array.

Reply:
Just filled it in with an actual implementation.